packages/node-ws/src/index.ts

```diff
@@ -49,8 +68,11 @@ export const createNodeWebSocket = (init: NodeWebSocketInit): NodeWebSocket => {
           await next()
           return
         }
-        const events = await createEvents(c)
-        wss.on('connection', (ws) => {
+
+        ;(async () => {
+          const events = await createEvents(c)
+          const ws = await nodeUpgradeWebSocket(c.env.incoming)
```

Comment:
What do you think about accepting `{ server: { incoming: request, outgoing: undefined } }` as Env?

When user use env as custom value, WebSocket helper may not work.

References: https://github.com/honojs/hono/issues/2645, https://github.com/honojs/hono/issues/2696, https://github.com/honojs/hono/pull/2812, https://github.com/honojs/hono/pull/2595#discussion_r1589946383.

Reply:
First, there are many theories about the implementation of `{ server: { incoming: request, outgoing: undefined } }`, but I consider this a minimal change to fix a bug.

Secondly, this is just my subjective view.
I see env as an area to store values given by the runtime.
Rather than storing values in the env hypothetically, I think a Variables area and middleware is a more natural way of doing this.

Again, I am not familiar with HonoJS core ideas, so this is subjective.

Reply:
@nakasyou @inaridiy 

Accepting `{ server: { incoming: request, outgoing: undefined }` for Env is a good idea! But, as @inaridiy said, we don't have to add the change about it to this PR. And you can create another PR after merging this. Shall we go with it?

Reply:
What do you do about this problem? Once we have discussed how to use env, should I reflect that discussion?

@nakasyou @yusukebe

Reply:
Let's merge this first without including Env things!